### DISCUSSION TITLE
How to use Chakra component with styled API by passing custom props

### DISCUSSION THREAD

**Discussion by anki247** [chakra-ui/chakra-ui#2704] [Q&A] [answered]
I am getting following error/warning
>Warning: Received `false` for a non-boolean attribute `highlight`.
If you want to write it to the DOM, pass a string instead: highlight="false" or highlight={value.toString()}.
If you used to conditionally omit it with highlight={condition && value}, pass highlight={condition ? value : undefined} instead.

by having this for example:
```js
const StyledBox = styled(Box)<BoxProps & {  highlight:  boolean }>`
  background-color:  ${({ highlight }) => highlight ? 'red' : 'white'}
`;
```

it seems that `styled()` in combination with chakra component did not filter the custom props and passing it to the underlying react div...
If I use `styled.div` instead the error does not appear

**Comment by with-heart** [comment, marked as answer]
Hi there @anki247! That's not a supported method of using Chakra components. If you'd like to use functions to create components, you can use [the `chakra` factory function](https://chakra-ui.com/docs/features/chakra-factory), but we have no supported way to write css inside template strings.

**Comment by anki247** [comment]
@with-heart Now I am facing again some similar issue with `const MotionBox = motion.custom(Box);` which is described [Chakra UI + Framer Motion doc](https://chakra-ui.com/guides/integrations/with-framer)
The props of motion (eg.: `initial, animate, variants, whileTap, whileHover, ...`) gets pasted to DOM and react complains about it...  

by having this 
```js 
      <MotionBox
          height="40px"
          bg="red.300"
          initial={false}
          animate="show"
          variant={{
            'test': { opacity: 1 }
          }}
          whileHover={{ scale: 1.1 }}
          whileTap={{ scale: 0.9 }}
        />
```

**DOM result** and the same warning/error log
`<div animate="show" variant="[object Object]" whilehover="[object Object]" whiletap="[object Object]" class="css-1kzx93d" style="transform: none;"></div>`

I have an example on [codesandbox](https://codesandbox.io/s/chakra-framer-motion-prop-warning-s1e4l?file=/src/index.js)
Do have a solution for this? Thx

**Reply by with-heart** [reply]
@anki247 Not yet but we've been discussing a change in #2717 that will allow us to fix this.
